src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -221,12 +181,12 @@ public bool TryGetTelemetryId(out Guid telemetryId)
                         return null;
                     }
 
-                    return filteredSets.Value.Select(s => ConvertToSuggestedActionSet(s)).WhereNotNull();
+                    return filteredSets.Value.SelectAsArray((s, arg) => ConvertToSuggestedActionSet(s, arg.Owner, arg.SubjectBuffer), (state.Target.Owner, state.Target.SubjectBuffer)).WhereNotNull();
```

Comment:
nit: my pref would be to just pass the state object along, so there's less unpacking/packing.  but this is also good given that it passes minimal data around.